### PR TITLE
Address Component UI Rework

### PR DESCRIPTION
**Pull request by jubalm**
This PR contains an updated address ui implementation in a number of places as this directly replaces the `SmallAddress` component and should retain all current features.

| Screenshot 2024-10-14 at 1 00 50 PM | Screenshot 2024-10-14 at 1 00 34 PM |
|--|--|
| Screenshot 2024-10-17 at 3 44 09 PM | |

- [x] hover should show full text preview
- [x] user feedback on successful copy

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]
Approved, though please make the recommended changes or reply to them with details why you disagree before merging.

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { CSSProperties, JSX } from 'preact/compat'
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: moved to `preact/jsx-runtime` in [62929f0](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/62929f0b9021b3289893abbfa617193b546302d0) (auto-add imports from IDE failed this time)

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	const dispatchEditCallback = () => {
+		editCallback?.(label)
+	}
+
+	useSignalEffect(() => {
+		if (copyStatus.value !== true) return
+		setTimeout(() => {
+			copyStatus.value = false
+		}, 1500)
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: Included [62929f0](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/62929f0b9021b3289893abbfa617193b546302d0)

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+export const InlineCard = ({ icon: Icon, label, editCallback, style }: InlineCardProps) => {
+	const copyStatus = useSignal(false)
+
+	const copyTextToClipboard = async (event: Event) => {
+		if (event.currentTarget instanceof HTMLButtonElement) {
+			event.currentTarget.blur()
+		}
+		await clipboardCopy(label)
+		copyStatus.value = true
+
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: Removed the extra line in [62929f0](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/62929f0b9021b3289893abbfa617193b546302d0)

File: app/css/interceptor.css [hunk 4]
```diff
@@ -1575,3 +1576,121 @@ header:has(form[role='search']) h1 {
...
+					svg { display: none }
+					span { display: inline-block }
+				}
+			}
+		}
+	}
+
+	[role=status] {
+		grid-area: start / start / end / end;
+		background: hsl(0deg 0% 19% / 93%);
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/css/interceptor.css]: was going to use `hsl` to add some color variance (saturation) as css variables but isn't really used here so moved the static colors to local (double-slash) variables

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		<>
+			<span class = 'inline-card' role = 'figure' style = { style } title = { label }>
+				<span role = 'img'><Icon /></span>
+				<data class = 'truncate text-legible' value = { label }>{label}</data>
+				<span role = 'menu'>
+					<button type = 'button' onClick = { copyTextToClipboard } tabIndex = { 1 }>
+						<span role = 'img'><Icon /></span>
+						<span><data class = 'truncate text-legible' value = { label }>{label}</data></span>
+						<span>
+							<svg width = '1em' height = '1em' viewBox = '0 0 24 24' fill = 'none' xmlns = 'http://www.w3.org/2000/svg'><path d = 'M14.188 4.813H4.813v9.375h.937V17H3.875A1.875 1.875 0 0 1 2 15.125V3.875C2 2.839 2.84 2 3.875 2h11.25C16.16 2 17 2.84 17 3.875V5.75h-2.812z' fill = 'currentColor' /><path fill-rule = 'evenodd' clip-rule = 'evenodd' d = 'M7 20V9a2 2 0 0 1 2-2h11a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H9a2 2 0 0 1-2-2m12-1h-9v-9h9z' fill = 'currentColor' /></svg>
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: moved the inline svg icons to `subcomponents/icons`

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import { CSSProperties, JSX } from 'preact/compat'
+import { useSignal, useSignalEffect } from '@preact/signals'
+import { clipboardCopy } from './clipboardcopy.js'
+
+export type InlineCardProps = {
+	icon: () => JSX.Element
+	label: string,
+	style?: CSSProperties,
+	editCallback?: (text: string) => void
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: Applied in [62929f0](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/62929f0b9021b3289893abbfa617193b546302d0)

File: app/ts/components/subcomponents/address.tsx [hunk 7]
```diff
@@ -6,6 +6,7 @@ import { AddressBookEntries, AddressBookEntry } from '../../types/addressBookTyp
 import { Website } from '../../types/websiteAccessTypes.js'
 import { Blockie } from './SVGBlockie.js'
 import { ComponentChildren } from 'preact'
+import { InlineCard } from './InlineCard.js'
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/address.tsx]: sorted `npm` library first then local files from farthest (`../../filefromparent`) to closest (`./sibling`) in the file tree

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export const InlineCard = ({ icon: Icon, label, editCallback, style }: InlineCardProps) => {
+	const copyStatus = useSignal(false)
+
+	const copyTextToClipboard = async (event: Event) => {
+		if (event.currentTarget instanceof HTMLButtonElement) {
+			event.currentTarget.blur()
+		}
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: used `JSX` import from `preact/jsx-runtime` to fix the type in [62929f0](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/62929f0b9021b3289893abbfa617193b546302d0)

File: app/ts/components/subcomponents/InlineCard.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import { useSignal, useSignalEffect } from '@preact/signals'
+import { clipboardCopy } from './clipboardcopy.js'
+import { CheckIcon, CopyIcon, EditIcon } from './icons.js'
+
+export type InlineCardProps = {
+	icon: () => JSX.Element
+	label: string
+	value?: string
+	style?: JSX.CSSProperties
+	onEditClick?: JSX.MouseEventHandler<HTMLButtonElement>
```
Comment (MicahZoltu) [inline app/ts/components/subcomponents/InlineCard.tsx:11]:
```suggestion
	onEditClicked?: JSX.MouseEventHandler<HTMLButtonElement>
```

Comment (jubalm) [reply app/ts/components/subcomponents/InlineCard.tsx]: change included in [43b1a77](https://github.com/DarkFlorist/TheInterceptor/pull/1156/commits/43b1a77124c491f70d7cb85ce1d50385f93e9a3b)

**Pull request merged**